dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxws.3.2/src/org/apache/cxf/jaxws/ServiceImpl.java

```diff
@@ -177,13 +192,17 @@ private void initializePorts() {
             }
 
             Map<String, Port> wsdlports = CastUtils.cast(serv.getPorts());
+            if(isFinestEnabled)  {
+                LOG.finest("WSDL ports size: " + wsdlports.size());
```

Comment:
Is this NPE safe?

Reply:
Following for loop address to wsdlports.values() too. So far we didn't get any issues from this. However, I'll make it safer for both wsdlports.size and wsdlports.values().